src/language-js/utils.js

```diff
@@ -107,11 +118,832 @@ function getLeftSidePathName(path, node) {
+  //   test: Identifier(A),
+  //   consequent: Identifier(B),
+  //   alternate: ConditionalExpression {
+  //     test: Identifier(C),
+  //     consequent: Identifier(D),
+  //     alternate: ConditionalExpression {
+  //       test: Identifier(E),
+  //       consequent: ConditionalExpression {
+  //         test: Identifier(F),
+  //         consequent: Identifier(G),
+  //         alternate: Identifier(H),
+  //       },
+  //       alternate: Identifier(I),
+  //     }
+  //   }
+  // }
+  //
+  // we should return this Array:
+  //
+  // [
+  //   Identifier(A),
+  //   Identifier(B),
+  //   Identifier(C),
+  //   Identifier(D),
+  //   Identifier(E),
+  //   Identifier(F),
+  //   Identifier(G),
+  //   Identifier(H),
+  //   Identifier(I)
+  // ];
+  //
+  // This loses the information about whether each node was the test,
+  // consequent, or alternate, but we don't care about that here- we are only
+  // flattening this structure to find if there's any JSXElements inside.
+  const nonConditionalExpressions = [];
+
+  function recurse(node) {
+    if (node.type === "ConditionalExpression") {
+      recurse(node.test);
+      recurse(node.consequent);
+      recurse(node.alternate);
+    } else {
+      nonConditionalExpressions.push(node);
+    }
+  }
+  recurse(node);
+
+  return nonConditionalExpressions;
+}
+
+function conditionalExpressionChainContainsJSX(node) {
+  return Boolean(getConditionalChainContents(node).find(isJSXNode));
+}
+
+function rawText(node) {
+  return node.extra ? node.extra.raw : node.raw;
+}
+
+function identity(x) {
+  return x;
+}
+
 module.exports = {
-  getLeftSide,
   getLeftSidePathName,
   hasNakedLeftSide,
-  hasNode,
   hasFlowShorthandAnnotationComment,
-  hasFlowAnnotationComment
+  hasFlowAnnotationComment,
+  isLiteral,
+  isNumericLiteral,
+  isStringLiteral,
+  isObjectType,
+  isFunctionOrArrowExpression,
+  isJSXNode,
+  isTheOnlyJSXElementInMarkdown,
+  isJSXWhitespaceExpression,
+  isMemberExpressionChain,
+  isGetterOrSetter,
+  isObjectTypePropertyAFunction,
+  isFunctionNotation,
+  isTypeAnnotationAFunction,
+  isBinaryish,
+  isMemberish,
+  isNodeStartingWithDeclare,
+  isSimpleFlowType,
+  isTestCall,
+  hasLeadingComment,
+  hasTrailingComment,
+  hasDanglingComments,
+  isCallOrOptionalCallExpression,
+  hasNgSideEffect,
+  isNgForOf,
+  isSimpleTemplateLiteral,
+  getFlowVariance,
+  classPropMayCauseASIProblems,
+  classChildNeedsASIProtection,
+  getTypeScriptMappedTypeModifier,
+  hasNewlineBetweenOrAfterDecorators,
+  isMeaningfulJSXText,
+  matchJsxWhitespaceRegex,
+  isEmptyJSXElement,
+  hasPrettierIgnore,
+  isFunctionCompositionFunction,
+  isLastStatement,
+  isFlowAnnotationComment,
+  hasLeadingOwnLineComment,
+  returnArgumentHasLeadingComment,
+  isStringPropSafeToCoerceToIdentifier,
+  isJestEachTemplateLiteral,
+  isTemplateOnItsOwnLine,
+  needsHardlineAfterDanglingComment,
+  conditionalExpressionChainContainsJSX,
+  rawText,
+  identity
```

Comment:
sort by name

Reply:
Defined order? Alphabetical order?

Reply:
yes.. whatever VS Code is doing for sorting lines :)